homeassistant/components/backup/__init__.py

```diff
@@ -36,22 +43,20 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Backup integration."""
-    hass.data[DOMAIN] = backup_manager = BackupManager(
-        hass, CoreBackupReaderWriter(hass)
-    )
-    await backup_manager.async_setup()
-
     with_hassio = is_hassio(hass)
 
-    async_register_websocket_handlers(hass, with_hassio)
+    if not with_hassio:
+        backup_manager = BackupManager(hass, CoreBackupReaderWriter(hass))
+    else:
+        # pylint: disable-next=import-outside-toplevel, hass-component-root-import
+        from homeassistant.components.hassio.backup import SupervisorBackupReaderWriter
```

Comment:
Do we need to add the hassio integration as an after dependency to the backup integration?

Reply:
Yeah, I've done that. I forgot it in the initial commit though